allennlp/nn/activations.py

```diff
@@ -53,13 +54,32 @@ def __call__(self, tensor: torch.Tensor) -> torch.Tensor:
         raise NotImplementedError
 
 
+class _ActivationLambda(torch.nn.Module):
+    """Wrapper around non PyTorch, lambda based activations to display them as modules whenever printing model."""
+
+    def __init__(self, func: Callable[[torch.Tensor], torch.Tensor], name: str):
+        super().__init__()
+        # Override class name to match activation name as PyTorch uses that field to create the textual
+        # representation.
+        self.__class__.__name__ = name
+        self.func = func
+
```

Comment:
This actually doesn't work:

```python
>>> class A:
>>>   def __init__(self, name):
>>>     self.__class__.__name__ = name
>>> 
>>> a = A('test')
>>> b = A('test2')
>>> a.__class__.__name__
'test2'
```

Looks like there's a `_get_name` function that we need to override instead.  How does this look?

```suggestion
        self._name = name
        self.func = func

    @overrides
    def _get_name(self):
        return self._name
```

Reply:
Yep, u're right.
https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/module.py#L1129

I'll change that and write better test (since we override there private class method - to address some pytorch changes in the future).